tribe-ext-export-mec-events.php

```diff
@@ -0,0 +1,362 @@
+		}
+
+		/**
+		 * If the button is clicked, start working
+		 *
+		 */
+		public function listen_for_export_button() {
+			if ( empty( $_REQUEST['export_events'] ) && empty( $_REQUEST['export_organizers'] ) && empty( $_REQUEST['export_venues'] ) || ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_events' ) && ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_organizers' ) && ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_venues' ) ) {
+				return;
+			} elseif ( ! empty( $_REQUEST['export_events'] ) || wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_events' ) ) {
+				$this->events_csv_setup();
+			} elseif ( ! empty( $_REQUEST['export_organizers'] ) || wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_organizers' ) ) {
+				$this->organizers_csv_setup();
+			} elseif ( ! empty( $_REQUEST['export_venues'] ) || wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_venues' ) ) {
+				$this->venues_csv_setup();
+			}
+		}
+
+		/**
+		 * Get the event data from Modern Events Calendar
+		 *
+		 * @return array $event_data
+		 */
+		public function get_event_data() {
+
+			$event_data = array();
+
+			global $wpdb;
+
+			$events = $wpdb->get_results( "SELECT t1.post_title,
+       				t1.post_content,
+       				t2.start,
+       				t2.end,
+       				t2.time_start,
+       				t2.time_end,
+       				t3.meta_value AS organizer_id,
+       				t4.meta_value AS venue_id,
+
+  					(SELECT sum(meta_value)
+   						FROM wp_postmeta AS t5
+   						WHERE t1.ID = t5.post_id
+     						AND t5.meta_key = 'mec_cost') AS cost
+				  FROM `wp_posts` AS t1
+				  INNER JOIN wp_mec_events AS t2
+				  INNER JOIN wp_postmeta AS t3
+				  INNER JOIN wp_postmeta AS t4
+				  WHERE t1.ID = t2.post_id
+  					AND t1.ID = t3.post_id
+  					AND t1.ID = t4.post_id
+  					AND t3.meta_key = 'mec_organizer_id'
+  					AND t4.meta_key = 'mec_location_id'" );
+
+			foreach ( $events as $event ) {
+				$row    = array();
+				$row[0] = $event->post_title;
+				$row[1] = $event->post_content;
+				$row[2] = ( $event->start == '0000-00-00' ? '' : $event->start );
+				$row[3] = ( $event->end == '0000-00-00' ? $event->start : $event->end );
+				$row[4] = gmdate( 'H:i:s', $event->time_start );
+				$row[5] = gmdate( 'H:i:s', $event->time_end );
+				$row[6] = ( get_term( $event->venue_id )->name != 'Uncategorized' ? get_term( $event->venue_id )->name : '' );
+				$row[7] = ( get_term( $event->organizer_id )->name != 'Uncategorized' ? get_term( $event->organizer_id )->name : '' );
+				$row[8] = $event->cost;
+
+				$event_data[] = $row;
+			}
+
+			return $event_data;
+		}
+
+		/**
+		 * Get the organizer data from Modern Events Calendar
+		 *
+		 * @return array $organizer_data
+		 */
+		public function get_organizer_data() {
+			$organizer_data = array();
+
+			global $wpdb;
+			/**
+			 * Get the organizers data from Modern Events Calendar and stores it in a variable.
+			 */
+			$organizers = $wpdb->get_results( "
+				SELECT DISTINCT meta_value 
```

Comment:
No need to change, but a note for future - a good way to format queries like this is:

```php
$wpdb->get_results( "
    SELECT *
    FROM   $wpdb->table
    WHERE  1
" );
```

That is, the closing `" );` is on its own line and is indented so as to line up with the opening part of the statement.